fieldtracing/fieldtracing.cpp

```diff
@@ -1221,287 +1027,271 @@ namespace FieldTracing {
+                  MPI_Allreduce(smallCellBWRunningDistance.data(), smallReducedCellBWRunningDistance.data(), smallSizeBW, MPI_FLOAT, MPI_MAX, MPI_COMM_WORLD);
+               }
+               for(int n=0; n<smallSizeFW; n++) {
+                  cellFWTracingStepSize[indicesToReduceFW[n]] = smallReducedCellFWTracingStepSize[n];
+                  cellFWRunningDistance[indicesToReduceFW[n]] = smallReducedCellFWRunningDistance[n];
+                  cellNeedsContinuedFWTracing[indicesToReduceFW[n]] = smallReducedCellNeedsContinuedFWTracing[n];
+                  cellFWConnection[indicesToReduceFW[n]] = smallReducedCellFWConnection[n];
+                  cellFWTracingCoordinates[indicesToReduceFW[n]] = smallSumCellFWTracingCoordinates[n];
+               }
+               for(int n=0; n<smallSizeBW; n++) {
+                  cellBWTracingStepSize[indicesToReduceBW[n]] = smallReducedCellBWTracingStepSize[n];
+                  cellBWRunningDistance[indicesToReduceBW[n]] = smallReducedCellBWRunningDistance[n];
+                  cellNeedsContinuedBWTracing[indicesToReduceBW[n]] = smallReducedCellNeedsContinuedBWTracing[n];
+                  cellBWConnection[indicesToReduceBW[n]] = smallReducedCellBWConnection[n];
+                  cellBWTracingCoordinates[indicesToReduceBW[n]] = smallSumCellBWTracingCoordinates[n];
                }
-               anyCellNeedsTracing = false;
             }
             #pragma omp barrier
             phiprof::stop("MPI-loop");
-            #pragma omp for schedule(dynamic) reduction(||:anyCellNeedsTracing)
+            #pragma omp single
+            {
+               cellsToDoFullBox = 0;
+               cellsToDoFluxRopes = 0;
+               // These are used for the small arrays ~70 lines up.
+               storedCellNeedsContinuedFWTracing = cellNeedsContinuedFWTracing;
+               storedCellNeedsContinuedBWTracing = cellNeedsContinuedBWTracing;
+            }
+            #pragma omp for schedule(dynamic) reduction(+:cellsToDoFullBox) reduction(+:cellsToDoFluxRopes)
             for(int n=0; n<globalDccrgSize; n++) {
-               if(reducedCellNeedsContinuedFWTracing[n] > 0) {
-                  anyCellNeedsTracing=true;
-                  cellNeedsContinuedFWTracing[n] = 1;
-                  
-                  // Update that nodes' tracing coordinates
-                  cellFWTracingCoordinates[n][0] = sumCellFWTracingCoordinates[n][0] / reducedCellNeedsContinuedFWTracing[n];
-                  cellFWTracingCoordinates[n][1] = sumCellFWTracingCoordinates[n][1] / reducedCellNeedsContinuedFWTracing[n];
-                  cellFWTracingCoordinates[n][2] = sumCellFWTracingCoordinates[n][2] / reducedCellNeedsContinuedFWTracing[n];
-                  
-                  cellFWRunningDistance[n] = reducedCellFWRunningDistance[n];
+               if(cellNeedsContinuedFWTracing[n]) {
```

Comment:
perhaps a nitpick, but this actually doesn't now consider how many cells are unprocessed, but rather, how many half-traces (FW or BW) are still unprocessed.

 So perhaps it should rather check for each cell if
`cellNeedsContinuedBWTracing[n]!=0` and `cellNeedsContinuedFWTracing[n]!=0` for `cellsToDoFullBox++` and inside that if:
 `cellBWConnection[n] == cellFWConnection[n]== TracingLineEndType::UNPROCESSED`
for `cellsToDoFluxRopes++`?

Reply:
OK the naming is confusing, but so far the fractions were of traces (2 per cell) and not cells. Should we make it thresholds of fractions of cells that have come to a valid result instead?

Reply:
Or rename the variables to tracesToDo...?